proposals/namespace_editing/README.md

```diff
@@ -0,0 +1,480 @@
+
+The UsdNamespaceEditor class will provide the following editing operations that work on object paths:
+
+* Delete[Prim/Property]AtPath(stage, path)
+	+ Authors/deletes **all scene description in the layer stack of the current edit target necessary** to remove the prim or property at path from the composed stage.
+
+* Move[Prim/Property]AtPath(stage, currentPath, newPath)
+	+ Authors/deletes all scene description in the layer stack of the current edit target necessary to make the composed prim or property at the currentPath instead exist at newPath.
+
+Additionally, the class will provide operations that take UsdPrim or UsdProperty objects for additional convenience. For each of the following XXXPrim functions that take a UsdPrim, there will be an equivalent XXXPropertry function that takes a UsdProperty
+
+* DeletePrim(usdPrim)
+	+ Authors/deletes all scene description in the layer stack of the current edit target necessary to remove the usdPrim from its composed stage.
+
+* RenamePrim(usdPrim, newName)
+	+ Authors all scene description in the layer stack of the current edit target necessary to change the name of the composed usdPrim to newName. The prim will retain its original parent prim.
+
+* ReparentPrim(usdPrim, newPrimParent)
+	+ Authors all scene description in the layer stack of the current edit target necessary to move the composed usdPrim from its original path to instead be a child of the composed prim, newPrimParent.
+
+* ReparentPrim(usdPrim, newPrimParent, newName)
+	+ Overload of ReparentPrim that also renames the composed prim to newName when moving it be a child of newParentPrim.
+
+For each of the above functions on both paths and UsdObjects, we will provide a corresponding Can[Operation] function (e.g. CanDeletePrimAtPath, CanMovePrim, CanRenameProperty) that returns whether the operation can be successfully performed with the current edit settings (see Authoring Settings below).
+
+#### Change Notifications
+
+Because namespace operations performed through the UsdNamespaceEditor API have a clear intent, we will be able to add the information about whether a UsdObject has been deleted, moved or renamed to UsdNotice::ObjectsChanged when edits are made through this API. In addition to the existing ObjectsChanged queries of `GetResyncedPaths()` and `GetChangedInfoOnlyPaths()`, we will add queries along the lines of `GetRenamedPaths()`, `GetMovedPaths()`, and `GetDeletedPaths()` to return what objects have been specifically edited using the UsdNamespaceEditor. We'll also provide the equivalent functions of `RenamedObject(UsdObject)`, `MovedObject(UsdObject)`, and `DeletedObject(UsdObject)` for querying a specific object as to whether it was namespace edited. Clients can make use of this new notice information to respond to changes a more efficient manner for namespace changes they know how to handle.
```

Comment:
Just thinking about the overall use of thee change notifications, and this may be outside the scope of the proposal, but have you also considered `GetAddedPaths()`.  In some circumstances this can remove the onus of keeping a stage prim path cache on the client side.  It may not be the same kind of namespace edit explicitly handled by the `UsdNamespaceEditor`, but feels like it might be in the same ballpark.

Reply:
It's conceivable to add `GetAddedPaths` but that's definitely outside the scope of this proposal. One advantage of having all our proposed namespace editing API in a specific new class is that we can justify tying the *new* notification about deleted and moved USD objects directly to the use of this *new* API without having to promise that we'd parse Sdf layer changes to determine delete or move intention. After typing this I'm realizing that it may make sense to have these notification functions be more explicit in that intent, i.e. `GetNamespaceEditDeletedPaths` and `GetNamespaceEditMovedPaths` or something like that.

The addition of `GetAddedPaths` would be something that should be filed as a separate request.